src/components/EditOrderForm.js

```diff
@@ -34,6 +34,16 @@ export function EditOrderForm({
 }) {
   const { user } = useContext(AuthContext);
   const [inputType, setInputType] = useState('datetime-local');
+  const customFeeRef = useRef(null);
+
+  useEffect(() => {
+    //Focus the customDeliveryFee input when hasExtraFee is true
+    if (haveExtraFee) {
+      setTimeout(() => {
+        customFeeRef.current.focus();
+      }, 250);
+    }
+  }, [haveExtraFee]);
```

Comment:
if the autofocus works, you can throw the useeffect and useref away

Reply:
I guess i over engineered this one 😅